### DISCUSSION TITLE
`bg-<name>-<number>/<opacity>` on hover

### DISCUSSION THREAD

**Discussion by smithliam** [tailwindlabs/tailwindcss#8685] [Ideas]
**What version of Tailwind CSS are you using?**

v3.1.3

**What build tool (or framework if it abstracts the build tool) are you using?**

SvelteKit

**What version of Node.js are you using?**

v16.14.0

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows 11

**Reproduction URL**

https://play.tailwindcss.com/8Z2ZdwAHiP

**Describe your issue**

This is the old way of doing `bg` + `bg-opacity`:

```html
<div class="bg-gray-900 bg-opacity-50"></div>

<!-- CSS output -->
<style>
  .bg-opacity-50 {
      --tw-bg-opacity: 0.5;
  }
  
  .bg-gray-900 {
      --tw-bg-opacity: 1;
      background-color: rgb(17 24 39 / var(--tw-bg-opacity));
  }
</style>
```

This is the short way of doing the same thing:

```html
<div class="bg-gray-900/50"></div>

<!-- CSS output -->
<style>
  .bg-gray-900\/50 {
      background-color: rgb(17 24 39 / 0.5);
  }
</style>
```

As you can see, the `--tw-bg-opacity` var didn't used in the second example. What if I want to change the opacity on `hover`?

```html
<div class="bg-gray-900/50 hover:bg-opacity-100"></div>

<style>
  .bg-gray-900\/50 {
      background-color: rgb(17 24 39 / 0.5);
  }
  .hover\:bg-opacity-100:hover {
      --tw-bg-opacity: 1;
  }
</style>
```

It isn't going to work!

**Proposed solution**

Do the same CSS output in the second example too.

That's all. Thanks

**Comment by adamwathan** [comment]
I would just do this:

```html
<div class="bg-gray-900/50 hover:bg-gray-900/100"></div>
```

I don't think we're going to change this I'm afraid so would really recommend using the strategy above 👍

**Reply by smithliam** [reply]
Why not? it's a problem.

**Reply by brandonmcconnell** [reply]
I mostly agree with @adamwathan's, and that's how we handle background opacity in our codebase, though I do see a strong case to be made for arbitrarily making an element's background translucent regardless of what bg color it uses when not hovered.

An *actual* workable solution if you need something custom would be to use a pseudo-element for the background and change the entire pseudo's opacity, like this (live demo on [Tailwind Play](https://play.tailwindcss.com/t6dNXofC9G)):
```html
<div class="relative before:absolute before:inset-0 before:w-full before:h-full before:-z-10 before:bg-red-500 hover:before:opacity-50">Some text</div>
```

Again, this isn't really ideal, so I would second the need for an updated spec for this. Until then, this is probably your best bet. You could also bundle this all into a new utility.

**Reply by babakfp** [reply]
I would like to do this, but with the current system, it's not possible.

```CSS
[class*=backdrop-blur] { @apply not-supports-backdrop-blur:bg-opacity-95 }
```

I don't like to add a long class to every single element that I use in the `backdrop-blur` class. I will also like to have some consistency between classes. There is no problem that not having the variable solves! So why should we create problems in exchange for nothing?

@adamwathan
